rust/lance-io/src/object_store/providers/memory.rs

```diff
@@ -29,7 +29,7 @@ impl ObjectStoreProvider for MemoryStoreProvider {
             max_iop_size: *DEFAULT_MAX_IOP_SIZE,
             use_constant_size_upload_parts: false,
             list_is_lexically_ordered: true,
-            io_parallelism: get_num_compute_intensive_cpus(),
+            io_parallelism: DEFAULT_CLOUD_IO_PARALLELISM,
```

Comment:
Why do we want to change this behavior? The original one seems more reasonable for memory store